core/trino-main/src/main/java/io/trino/sql/planner/LiteralInterpreter.java

```diff
@@ -65,31 +65,34 @@
 
 public final class LiteralInterpreter
 {
-    private LiteralInterpreter() {}
+    private final PlannerContext plannerContext;
+    private final Session session;
+    private final ConnectorSession connectorSession;
+    private final InterpretedFunctionInvoker functionInvoker;
 
-    public static Object evaluate(PlannerContext plannerContext, Session session, Map<NodeRef<Expression>, Type> types, Expression node)
+    public LiteralInterpreter(PlannerContext plannerContext, Session session)
+    {
+        this.plannerContext = requireNonNull(plannerContext, "plannerContext is null");
+        this.session = requireNonNull(session, "session is null");
+        this.connectorSession = session.toConnectorSession();
+        this.functionInvoker = new InterpretedFunctionInvoker(plannerContext.getFunctionManager());
+    }
+
+    public Object evaluate(Map<NodeRef<Expression>, Type> types, Expression node)
```

Comment:
I don't think it has to be a map. It can just be `evaluate(Expression node, Type type)`

Reply:
I thought about that too.
Actually, i did such change locally:

- there are 4 call sites AFAIR; two provide a map; two provide explicit type
- for most literal types, we don't check the type at all, so picking up a map entry so that it's not used seemed wasteful
- to avoid that lookup one can introduce a `Function<Expression,Type>` type provider -- either looks up from a Map, or provides a known type
- however, constructing such lookup is most likely equally expensive as constructing an `ImmutableMap` with single entry, so it feels i am not saving anything

for these reasons, i scratched the change and didn't include it in the PR

Reply:
> however, constructing such lookup is most likely equally expensive as constructing an ImmutableMap with single entry, so it feels i am not saving anything

It's not about it being expensive. It's more about conveying false sense that we need a `Map` there (we don't)

Reply:
I agree. Note that  it's pre-existing and i didn't choose to change it here.
 i was concerned about performance implications of doing so, because -- as we learned -- this class is on a critical path for planner's performance.

Reply:
I've made a follow up PR: https://github.com/trinodb/trino/pull/11979. It's super simple.

> i was concerned about performance implications of doing so, because -- as we learned -- this class is on a critical path for planner's performance.

It should also improve performance as we don't have to lookup by String anymore in case of `BIGINTs` and other non-parametrized types (and also for parametrized `varchar`)

Reply:
> It should also improve performance as we don't have to lookup by String anymore in case of BIGINTs and othe

thanks @sopel39 , i didn't notice that